### PR TITLE
239 frontend new select exercise based on muscle group page

### PR DESCRIPTION
**Pull request by AaronFmyHub**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [x] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

Created the page where user select exercise based on prior muscle selection.
This handles the issue @elliotwesoff created before about backend query, and it closes another issue #239 
This page displays exercises that match user selected muscle group from the list we provided and further lower informational searching cost for user by having user friendly UI/UX design.
The page contains select/unselect feature, and the number next to Add activates when user has a valid selection, it indicates the total exercises user selected.
It also includes a reset feature, which is useful when users are dealing with large lists.

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots
[exerciseFilter.webm](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/146596967/32f3da58-c9c8-47cb-9f01-46552db79554)

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# [Optional] Are there any post-deployment tasks we need to perform?
We need to nav from muscleGroup to this page and pass in the passNext.

# AI Usage
ChatGPT explained Truthy/Falsy to me and helped me to find a solution display filtered content.
It also wrote the code to reset array, which contains user selections.
https://chat.openai.com/share/32926458-88e0-430c-a875-85edf02efb96

### CODE REVIEW

**Review by StefanoRubini — COMMENTED** [review]
Hello Aaron! Nice work for implementing the Select Exercise page. Could it be possible to add some styling to the right-hand side where it says "Add-x"? Thank you! 😃
Screenshot 2024-04-24 at 4 44 38 PM

File: apps/expo/src/app/filteredExercises/index.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react'
+import { Dimensions, Pressable, Text, TouchableOpacity, View } from 'react-native'
+import { ScrollView } from 'react-native-gesture-handler'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { router } from 'expo-router'
+
+import { muscleSelectUser } from '~/app/muscleGroup'
+import NavBar from '~/components/ui/nav-bar/NavBar'
```
Comment (bigboidanwithacan) [inline apps/expo/src/app/filteredExercises/index.tsx:8]: Remove unused import

**Review by bigboidanwithacan — APPROVED** [review]
Great Job on this Aaron! I checked out the branch and tested it on my end, and everything worked great!  
image

File: apps/expo/src/app/filteredExercises/index.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      } else {
+        // Negative value should not appear
+        setAddCount(addCount - 1)
+      }
+    }
+  }
+
+  return (
+    <SafeAreaView style={{ backgroundColor: '#1C1B1B', flex: 1 }}>
+      {/*<NavBar*/}
```
Comment (elliotwesoff) [inline apps/expo/src/app/filteredExercises/index.tsx:53]: Please remove commented code if it won't be used later

File: apps/expo/src/app/filteredExercises/index.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        <Pressable onPress={() => router.back()}>
+          {addCount == 0 ? (
+            <Text style={{ color: '#CACACA', fontSize: 20 }}>Add</Text>
+          ) : (
+            <Text style={{ color: '#CACACA', fontSize: 20 }}>Add{addCount}</Text>
+          )}
+        </Pressable>
+      </View>
+
+      {/*<View className='flex flex-row items-center'>*/}
```
Comment (elliotwesoff) [inline apps/expo/src/app/filteredExercises/index.tsx:78]: Please remove commented code

File: apps/expo/src/app/filteredExercises/index.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      {/*    }}*/}
+      {/*    resizeMode='contain'*/}
+      {/*  />*/}
+      {/*</View>*/}
+
+      <View className='pb-px-20 flex flex-row' style={{ height: 0.7 * screenHeight - 50 }}>
+        <ScrollView>
+          {data?.map((exercise: any) => (
+            <View key={exercise.id} className='flex flex-row rounded-lg p-2'>
+              {exercise?.bodyPart == 'ARMS' && muscle == muscleSelectUser.Arm && (
```
Comment (elliotwesoff) [inline apps/expo/src/app/filteredExercises/index.tsx:94]: From here to line 161 could be refactored into a loop, but I understand if we don't have time for that now

File: apps/expo/src/app/filteredExercises/index.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            </View>
+          ))}
+        </ScrollView>
+      </View>
+
+      <View className='m-2 flex flex-row items-center' style={{ position: 'absolute', bottom: 40 }}>
+        <TouchableOpacity
+          className='ml-5 mr-2 mt-1 flex-1 rounded-lg px-4 py-4 font-bold'
+          style={{ backgroundColor: '#CACACA' }}
+          onPress={() => {}}
```
Comment (elliotwesoff) [inline apps/expo/src/app/filteredExercises/index.tsx:172]: Should this onPress do something?

**Review by elliotwesoff — APPROVED** [review]
Just double check the onPress callback for the finish button. I left a comment. Everything else LGTM.

**Comment by taylorfinelli** [comment]
Isn't this supposed to filter based on the body part selected on the diagram?

**Review by taylorfinelli — CHANGES_REQUESTED** [review]
Needs to be connected to the 2D model page and filter workouts based on selected body part

**Comment by taylorfinelli** [comment]
Updated to include functionality and minor styling changes

https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/e7b6065a-ae61-436f-b7b8-58b9eff99d5f

Ready to be linked to workout view page. Go to `apps/expo/src/app/filteredExercises/index.tsx` and change the pathname on line 33

**Review by taylorfinelli — APPROVED** [review]

**Review by mva919 — APPROVED** [review]
🪨

**Pull request merged**